src/core/CCOpt.mli

```diff
@@ -9,9 +9,13 @@ val map : ('a -> 'b) -> 'a t -> 'b t
 (** Transform the element inside, if any *)
 
 val map_or : default:'b -> ('a -> 'b) -> 'a t -> 'b
-(** [map_or ~default f o] is [f x] if [o = Some x], [default otherwise]
+(** [map_or ~default f o] is [f x] if [o = Some x], [default] otherwise
     @since 0.16 *)
 
+val map_lazy : (unit -> 'b) -> ('a -> 'b) -> 'a t -> 'b
+(** [map_lazy default_fn f o] if [f o] if [o = Some x], [default_fn ()] otherwise 
+    @since 1.2 *)
```

Comment:
I tend to use `@since NEXT_RELEASE` in these comments. It's okay, because the next release will most likely be 1.2, but still.